python/tank/descriptor/io_descriptor/git_tag.py

```diff
@@ -212,35 +232,16 @@ def _get_latest_by_pattern(self, pattern):
                 "Git repository %s doesn't have any tags!" % self._path
             )
 
-        latest_tag = self._find_latest_tag_by_pattern(git_tags, pattern)
-        if latest_tag is None:
-            raise TankDescriptorError(
-                "'%s' does not have a version matching the pattern '%s'. "
-                "Available versions are: %s"
-                % (self.get_system_name(), pattern, ", ".join(git_tags))
-            )
-
-        return latest_tag
+        return git_tags
 
     def _get_latest_version(self):
         """
         Returns a descriptor object that represents the latest version.
         :returns: IODescriptorGitTag object
         """
-        try:
-            # clone the repo, find the latest tag (chronologically)
-            # for the repository, across all branches
-            commands = [
-                "for-each-ref refs/tags --sort=-creatordate --format='%(refname:short)' --count=1"
-            ]
-            latest_tag = self._tmp_clone_then_execute_git_commands(commands)
-
-        except Exception as e:
-            raise TankDescriptorError(
-                "Could not get latest tag for %s: %s" % (self._path, e)
-            )
-
-        if latest_tag == "":
+        tags = self._fetch_tags()
+        latest_tag = self._find_latest_tag_by_pattern(tags, pattern=None)
```

Comment:
This changes the functionality from the previous version. It may or may not matter, but previously we were sorting by date, now we are sorting by version. If the version numbers are done in a reasonable way you should end up with the same result and since it's not a public method it should be okay, but it's still worth noting the difference.

Reply:
Actually, thinking about it more, even though it isn't a public method, the behaviour could change if a user was just making tags with a description instead of version numbers and expect to get the latest one

Reply:
For posterity, here's a community post we made where people agreed with the change: https://community.shotgunsoftware.com/t/toolkits-git-descriptor-behavior-change/8390/14